### PR TITLE
PLAT-123306: [Agate] TemperatureControl: Arc does not look a normal shape

### PR DESCRIPTION
**Pull request by paul-beldean-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
- ArcSlider would not display correctly when 'min' was greater than 'max' or when 'min' and 'max' were equal

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- Made ArcSlider display correctly
- The change in ArcSlider.js was done to get a 'valueAngle' for the 'min' value to have as the default if 'min' is greater or equal to 'max', the same was done to have a minimum 'value' in ArcSliderBehaviorDecorator.js when 'min' is greater or equal to 'max'
- Added componentDidUpdate in ArcSliderBehaviorDecorator.js to set the value to be in range if it is bigger than 'max' or lower than 'min'
- Incremented 'max' in ArcSlider/utils.js to not have division by 0 if 'value', 'min' and 'max' are equal

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-123306

### Comments

Enact-DCO-1.0-Signed-off-by: Paul Beldean paul.beldean@lgepartner.com

### CODE REVIEW

**Review by daniel-stoian-lgp — APPROVED** [review]

File: ArcSlider/ArcSliderBehaviorDecorator.js [hunk 1]
```diff
@@ -145,6 +145,16 @@ const ArcSliderBehaviorDecorator = hoc((config, Wrapped) => {
 			};
 		}
 
+		componentDidUpdate (prevProps) {
+			if ( this.props.max !== prevProps.max || this.props.min !== prevProps.min) {
+				if (this.state.value && this.state.value >= this.props.max) {
+					this.handleChange(null, this.props.max);
+				} else if ( this.state.value && this.state.value <= this.props.min) {
+					this.handleChange(null, this.props.min);
```
Comment (jeonghee27) [inline ArcSlider/ArcSliderBehaviorDecorator.js:153]: how about use clamp function?

import {clamp} from '@enact/core/util';

File: tests/screenshot/apps/components/TemperatureControl.js [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
+	// Min value equals max value - [PLAT-123306]
+	<TemperatureControl
+		max={30}
+		min={30}
+	/>,
+	// Min value > max value - [PLAT-123306]
+	<TemperatureControl
+		max={30}
+		min={50}
+	/>
```
Comment (jeonghee27) [inline tests/screenshot/apps/components/TemperatureControl.js:14]: Please add normal cases also.

<TemperatureControl
		max={50}
		min={30}
	/>

File: ArcSlider/ArcSliderBehaviorDecorator.js [hunk 3]
```diff
@@ -180,7 +190,12 @@ const ArcSliderBehaviorDecorator = hoc((config, Wrapped) => {
 
 			const angle = positionToAngle(coordsInSvg, radius * 2 - strokeWidth);
 			// get the value based on the angle, min and max
-			let value = angleToValue(angle, min, max, startAngle, endAngle);
+			let value;
+			if (max <= min) {
+				value = angleToValue(angle, min, min, startAngle, endAngle);
+			} else {
+				value = angleToValue(angle, min, max, startAngle, endAngle);
+			}
```
Comment (jeonghee27) [inline ArcSlider/ArcSliderBehaviorDecorator.js:198]:
```suggestion
			const value = angleToValue(angle, min, Math.max(min, max), startAngle, endAngle);
```
How about it? There may be other good alternatives.

File: ArcSlider/ArcSlider.js [hunk 4]
```diff
@@ -210,7 +210,12 @@ const ArcSliderBase = kind({
 
 	render: ({'aria-valuetext': ariaValuetext, backgroundColor, componentRef, disabled, endAngle, foregroundColor, isFocused, max, min, radius, size, slotCenter, startAngle, strokeWidth, value, ...rest}, context) => {
 		const {accent: accentColor} = context || {};
-		const valueAngle = valueToAngle(value, min, max, startAngle, endAngle);
+		let valueAngle;
+		if (max <= min) {
+			valueAngle = valueToAngle(min, min, min, startAngle, endAngle);
+		} else {
+			valueAngle = valueToAngle(value, min, max, startAngle, endAngle);
+		}
```
Comment (jeonghee27) [inline ArcSlider/ArcSlider.js:218]: same with ArcSliderBehaviorDecorator.js

**Comment by jeonghee27** [comment]
There is tavis fail. please check it.

File: ArcSlider/ArcSliderBehaviorDecorator.js (hunk 1 above)
Comment (paul-beldean-lgp) [reply ArcSlider/ArcSliderBehaviorDecorator.js:153]: good idea

File: ArcSlider/ArcSliderBehaviorDecorator.js (hunk 3 above)
Comment (paul-beldean-lgp) [reply ArcSlider/ArcSliderBehaviorDecorator.js:198]: yes, it's better this way, but I will need to use "let" because the value is updated after this

**Review by jeonghee27 — APPROVED** [review]
LGTM

**Pull request merged**